torchaudio/prototype/ctc_decoder/ctc_decoder.py

```diff
@@ -257,3 +263,46 @@ def lexicon_decoder(
         sil_token=sil_token,
         unk_word=unk_word,
     )
+
+
+def download_pretrained_files(model: str) -> PretrainedFiles:
+    """
+    Retrieves pretrained data files used for CTC decoder.
+
+    Args:
+        model (str): pretrained language model to download
+            Options: ["librispeech-3-gram", "librispeech-4-gram", "librispeech"]
+
+    Returns:
+        Dictionary with the following key mapping:
+            lm: path corresponding to downloaded language model, or None if lm not provided
+            lexicon: path corresponding to downloaded lexicon file
+            tokens: path corresponding to downloaded tokens file
+    """
+
+    hub_dir = torch.hub.get_dir()
+    path = f"{hub_dir}/{model}"
+
+    url_prefix = "https://download.pytorch.org/torchaudio/decoder-assets"
+    model_dir = f"{url_prefix}/{model}"
```

Comment:
1. I think it is better to extract the logic which handles "key -> route on download" to a helper function.
2. Whether if lm is available (whether `lm` attribute should be `None`) should be handled by the code here. It should not be the result of successful download.


```python

routes = _get_route(model)
token_file, lexicon_file, lm_file = routes

download_url_to_file(<token_uri>, <local_token_path>)
download_url_to_file(<lexicon_uri>, <local_lexicon_path>)
if lm_file is not None:
    download_url_to_file(<lm_uri>, <local_lm_path>)

return ...
```